File: src/main/java/com/amaze/filemanager/adapters/HiddenAdapter.java
package com.amaze.filemanager.adapters;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;
import com.amaze.filemanager.R;
import com.amaze.filemanager.activities.MainActivity;
import com.amaze.filemanager.filesystem.BaseFile;
import com.amaze.filemanager.filesystem.HFile;
import com.amaze.filemanager.fragments.MainFragment;
import com.amaze.filemanager.services.DeleteTask;
import com.amaze.filemanager.utils.DataUtils;
import com.amaze.filemanager.utils.Futils;
import com.amaze.filemanager.utils.OpenMode;

import java.io.File;
import java.util.ArrayList;


/**
 * Created by Arpit on 16-11-2014.
 */
public class HiddenAdapter extends RecyclerArrayAdapter<HFile, HiddenAdapter.ViewHolder> {
    private Futils utils;

    private MainFragment context;
    private Context c;
    public ArrayList<HFile> items;
    private MaterialDialog materialDialog;
    private boolean hide;
    ///	public HashMap<Integer, Boolean> myChecked = new HashMap<Integer, Boolean>();

    public HiddenAdapter(Context context, MainFragment mainFrag, Futils utils, @LayoutRes int layoutId,
                         ArrayList<HFile> items, MaterialDialog materialDialog, boolean hide) {
        addAll(items);
        this.utils = utils;
        this.c = context;
        this.context = mainFrag;
        this.items = items;
        this.hide = hide;
        this.materialDialog = materialDialog;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater mInflater = (LayoutInflater) c
                .getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
        View view = mInflater.inflate(R.layout.bookmarkrow, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.render(position, getItem(position));
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        private ImageButton image;
        private TextView txtTitle;
        private TextView txtDesc;
        private LinearLayout row;

        ViewHolder(View view) {
            super(view);

            txtTitle = (TextView) view.findViewById(R.id.text1);
            image = (ImageButton) view.findViewById(R.id.delete_button);
            txtDesc = (TextView) view.findViewById(R.id.text2);
            row = (LinearLayout) view.findViewById(R.id.bookmarkrow);
        }

        void render(final int position, final HFile file) {
            txtTitle.setText(file.getName());
            String a = file.getReadablePath(file.getPath());
            txtDesc.setText(a);

            if (hide)
                image.setVisibility(View.GONE);

            // TODO: move the listeners to the constructor
            image.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (!file.isSmb() && file.isDirectory()) {
                        ArrayList<BaseFile> a = new ArrayList<>();
                        BaseFile baseFile = new BaseFile(items.get(position).getPath() + "/.nomedia");
                        baseFile.setMode(OpenMode.FILE);
                        a.add(baseFile);
                        new DeleteTask(context.getActivity().getContentResolver(), c).execute((a));
                    }
                    DataUtils.removeHiddenFile(items.get(position).getPath());
                    items.remove(items.get(position));
                    notifyDataSetChanged();
                }
            });
            row.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    materialDialog.dismiss();
                    new Thread(new Runnable() {
                        @Override
                        public void run() {
                            if (file.isDirectory()) {
                                context.getActivity().runOnUiThread(new Runnable() {
                                    @Override
                                    public void run() {
                                        context.loadlist(file.getPath(), false, OpenMode.UNKNOWN);
                                    }
                                });
                            } else {
                                if (!file.isSmb()) {
                                    context.getActivity().runOnUiThread(new Runnable() {
                                        @Override
                                        public void run() {
                                            utils.openFile(new File(file.getPath()), (MainActivity) context.getActivity());
                                        }
                                    });
                                }
                            }
                        }
                    }).start();
                }
            });
        }
    }

    public void updateDialog(MaterialDialog dialog) {
        materialDialog = dialog;
    }
}

File: app/src/main/java/com/amaze/filemanager/adapters/HiddenAdapter.kt
/*
 * Copyright (C) 2014-2020 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
 * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
 *
 * This file is part of Amaze File Manager.
 *
 * Amaze File Manager is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.amaze.filemanager.adapters

import android.app.Activity
import android.content.Context
import android.content.SharedPreferences
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.afollestad.materialdialogs.MaterialDialog
import com.amaze.filemanager.R
import com.amaze.filemanager.adapters.holders.HiddenViewHolder
import com.amaze.filemanager.asynchronous.asynctasks.DeleteTask
import com.amaze.filemanager.fileoperations.filesystem.OpenMode
import com.amaze.filemanager.filesystem.HybridFile
import com.amaze.filemanager.filesystem.HybridFileParcelable
import com.amaze.filemanager.filesystem.files.FileUtils
import com.amaze.filemanager.ui.activities.MainActivity
import com.amaze.filemanager.ui.fragments.MainFragment
import com.amaze.filemanager.utils.DataUtils
import java.io.File
import java.util.ArrayList
import kotlin.concurrent.thread

/**
 * This Adapter contains all logic related to showing the list of hidden files.
 *
 * @see com.amaze.filemanager.adapters.holders.HiddenViewHolder
 */
class HiddenAdapter(
    private val context: Context,
    private val mainFragment: MainFragment,
    private val sharedPrefs: SharedPreferences,
    hiddenFiles: List<HybridFile>,
    var materialDialog: MaterialDialog?,
    private val hide: Boolean
) : RecyclerView.Adapter<HiddenViewHolder>() {

    companion object {
        private const val TAG = "HiddenAdapter"
    }

    private val hiddenFiles = hiddenFiles.toMutableList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HiddenViewHolder {
        val mInflater = context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = mInflater.inflate(R.layout.bookmarkrow, parent, false)
        return HiddenViewHolder(view)
    }

    override fun onBindViewHolder(holder: HiddenViewHolder, position: Int) {
        val file = hiddenFiles[position]
        holder.textTitle.text = file.getName(context)
        holder.textDescription.text = file.getReadablePath(file.path)
        if (hide) {
            holder.deleteButton.visibility = View.GONE
        }
        holder.deleteButton.setOnClickListener {
            // if the user taps on the delete button, un-hide the file.
            // TODO: the "hide files" feature just hide files from view in Amaze and not create
            // .nomedia
            if (!file.isSmb && file.isDirectory(context)) {
                val nomediaFile = HybridFileParcelable(
                    hiddenFiles[position].path + "/" + FileUtils.NOMEDIA_FILE
                )
                nomediaFile.mode = OpenMode.FILE
                val filesToDelete = ArrayList<HybridFileParcelable>()
                filesToDelete.add(nomediaFile)
                val task = DeleteTask(context, false)
                task.execute(filesToDelete)
            }
            DataUtils.getInstance().removeHiddenFile(hiddenFiles[position].path)
            hiddenFiles.remove(hiddenFiles[position])
            notifyItemRemoved(position)
        }
        holder.row.setOnClickListener {
            // if the user taps on the hidden file, take the user there.
            materialDialog?.dismiss()

            thread {
                val fragmentActivity = mainFragment.requireActivity()
                if (file.isDirectory(context)) {
                    fragmentActivity.runOnUiThread {
                        mainFragment.loadlist(
                            file.path,
                            false,
                            OpenMode.UNKNOWN,
                            false
                        )
                    }
                } else if (!file.isSmb) {
                    fragmentActivity.runOnUiThread {
                        FileUtils.openFile(
                            File(file.path),
                            (fragmentActivity as MainActivity),
                            sharedPrefs
                        )
                    }
                }
            }
        }
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemCount(): Int = hiddenFiles.size
}
